### DISCUSSION TITLE
Query Regarding Correlation Identifier in MTConnect

### DISCUSSION THREAD

**Discussion by Pranay265** [TrakHound/MTConnect.NET#64] [General]
Hello @PatrickRitchie ,
 
I am reaching out to inquire about a specific feature in the new version of MTConnect. I am curious to know if there is any provision for a 'correlation' identifier within the new version of MTConnect.

Specifically, I am interested in understanding if there is a mechanism in place that allows for the association or correlation of different events, data items within the MTConnect framework. 

I did check in the latest version, i.e., 6.4.1, but I didn't find any correlation between events. If such a feature exists or if there are any plans to incorporate it in the future releases, I would greatly appreciate any insights or information you could provide on this matter. Looking forward to hearing from you soon.

Thanks !

**Comment by PatrickRitchie** [comment]
Thanks for the question @Pranay265, do you have an example of what you are looking for?

**Comment by Pranay265** [comment]
Sorry for the delay in response. To clarify my question about the correlation identifier, let me provide an example. The goal is to link various data items to a specific job for easier tracking and analysis.

**Example Scenario:**
Consider a machine performing multiple jobs. We want to track all events and data items related to a specific job. Here are some sample data items:

_Process State Data Item:_
DataItem name="CleaningWizard" id="cleaningWizard" category="EVENT" type="PROCESS_STATE" 

_Material Data Item:_
DataItem name="MaterialLoad" id="materialLoad1" category="EVENT" type="MATERIAL"

Current Challenge:
There is no explicit way to correlate these data items to a specific job. For example, we want to identify that both 'CleaningWizard' and 'MaterialLoad' belong to Job A.

Desired Feature:
We need a feature to associate events and data items with a specific job, such as through a unique job identifier or correlation ID.

**Comment by PatrickRitchie** [comment]
That is an interesting idea to explicitly link a job to a DataItem's Observation. I believe the way MTConnect is intended to accomplish this would be to organize the machine model so that each component(s) that can run a separate job is under its own Path component. 

Under each Path component, you can have a ProcessState, PartId, ProcessId, etc. to record what job each "machine process" is currently working on.

Instead of explicitly relating the job to each DataItem's Observation, you would look at the timestamp of when the job was loaded and associate all of the observations at that timestamp and after, with that job. This type of relationship (and other relationships) are usually handled in the application layer as opposed to the protocol since use cases can vary so much.

If you are interested in proposing this idea to be added to the MTConnect standard, I would recommend joining the free [MTConnect Standards Committee](https://www.mtconnect.org/standards-committee) (if you aren't already a member) and creating an issue for the ShopFloor Information working group to discuss. I would be interested to see what they would say. MTConnect recently added a "conditionId" to create a relationship among alarms so this would be a similar concept. https://model.mtconnect.org/#Structure___19_0_3_45f01b9_1579566531113_85883_25726

This is obviously something that can be handled in a variety of ways so if anyone else has anything to add, then please feel free to join the discussion.

**Comment by Pranay265** [comment]
Thank you for your detailed and insightful response. Yes, we are currently using the approach you mentioned, utilizing timestamps to associate observations with the corresponding job. However, explicitly linking a job to a DataItem's Observation via a correlation identifier could provide a more direct and potentially less error-prone way to manage and analyze job-specific data.

I appreciate the suggestion to join the MTConnect Standards Committee and propose this idea. It would be valuable to hear their thoughts on this potential enhancement.
